File: backend/app/schemas/connections.py
from typing import Optional

from pydantic import BaseModel, Field


class EdgePCConfig(BaseModel):
    host: str = Field(..., min_length=1)
    port: int = 22
    username: str = Field(..., min_length=1)


class MSSQLConfig(BaseModel):
    enabled: bool = False
    host: str = ""
    port: int = 1433
    username: str = ""
    password: Optional[str] = None
    database: str = "HISTORISCH"
    table: str = "Tab_Actual"
    poll_interval_seconds: int = 60
    window_minutes: int = 10
    max_rows_per_poll: int = 5000


class ConnectionsRead(BaseModel):
    edge_pc: Optional[EdgePCConfig] = None
    mssql: MSSQLConfig


class ConnectionsUpdate(BaseModel):
    edge_pc: Optional[EdgePCConfig] = None
    mssql: Optional[MSSQLConfig] = None


class MSSQLTestRequest(BaseModel):
    config: Optional[MSSQLConfig] = None


class MSSQLTestResponse(BaseModel):
    ok: bool
    message: str

File: frontend/src/api/connections.ts
import api from "./index";

export interface EdgePCConfig {
    host: string;
    port: number;
    username: string;
}

export interface MSSQLConfig {
    enabled: boolean;
    host: string;
    port: number;
    username: string;
    password?: string | null;
    database: string;
    table: string;
    poll_interval_seconds: number;
    window_minutes: number;
    max_rows_per_poll: number;
}

export interface ConnectionsRead {
    edge_pc?: EdgePCConfig | null;
    mssql: MSSQLConfig;
}

export interface ConnectionsUpdate {
    edge_pc?: EdgePCConfig | null;
    mssql?: Partial<MSSQLConfig> | null;
}

export interface MSSQLTestResponse {
    ok: boolean;
    message: string;
}

export const connectionsApi = {
    get: async (): Promise<ConnectionsRead> => {
        const { data } = await api.get("/connections");
        return data;
    },

    update: async (payload: ConnectionsUpdate): Promise<ConnectionsRead> => {
        const { data } = await api.put("/connections", payload);
        return data;
    },

    testMssql: async (payload?: { config?: MSSQLConfig }): Promise<MSSQLTestResponse> => {
        const { data } = await api.post("/connections/test/mssql", payload || {});
        return data;
    },
};
